internals-js/src/extractSubgraphsFromSupergraph.ts

```diff
@@ -328,15 +328,46 @@ export function extractSubgraphsFromSupergraph(supergraph: Schema): Subgraphs {
           }
           // Fall-through on purpose.
         case 'InputObjectType':
+          const typeApplications = type.appliedDirectivesOf(typeDirective);
+          // Note that we can have more that one `@join__type` for a given graph if there is multiple keys, so we collect
+          // the actual set of subgraphs defining the type.
+          const subgraphsDefiningType: Subgraph[] = [];
+          for (const app of typeApplications) {
+            const subgraph = subgraphs.get(graphEnumNameToSubgraphName.get(app.arguments().graph)!)!;
```

Comment:
I know that these type assertions should be theoretically fine, but I do worry slightly about hand edited supergraphs. Maybe convert these to use `assert`'s?

Reply:
I'll change that in #2655 as it centralises a bit the places where this is done, and would rather get that one in quickly.